engine/src/main/java/org/destinationsol/game/ObjectManager.java

```diff
@@ -100,15 +100,20 @@ public void update(SolGame game) {
 
             if (o.shouldBeRemoved(game)) {
                 removeObjDelayed(o);
+                if (game.getHero().isAlive() && game.getHero().isNonTranscendent() && o == game.getHero().getHero()) {
```

Comment:
Might make sense extracting `game.getHero()` to a local variable. Also, `game.getHero().getHero()` is the perfect example of how confusing naming the hero's ship `hero` is!